packages/ra-data-graphcool/README.md

```diff
@@ -87,6 +87,41 @@ Or supply your client directly with:
 buildGraphcoolProvider({ client: myClient });
 ```
 
+### Overriding a specific query
+
+The default behavior might not be optimized especially when dealing with references. You can override a specific query by wrapping the `buildQuery` function:
+
+```js
+// in src/dataProvider.js
+import buildGraphcoolProvider, { buildQuery } from 'ra-data-graphcool';
+
+const overrideBuildQuery = buildQuery => (fetchType, resource, params) => {
```

Comment:
I think the signature is wrong here: `(buildQuery) => (introspectionResults) => (fetchType, resource, params)`.

Same goes for `const builtQuery = buildQuery(introspectionResults)(fetchType, resource, params);`

Reply:
Right !

Reply:
It seems cumbersome to use it that way, isn't it? Lot of boilerplate again...

What do you think of the following usage:

```js
// in src/dataProvider.js
import buildGraphcoolProvider from 'ra-data-graphcool';

const buildQuery = defaultBuildQuery => (fetchType, resource, params) => {
    const builtQuery = defaultBuildQuery(fetchType, resource, params);

    if (resource === 'Command' && fetchType === 'GET_ONE') {
        return {
            // Use the default query variables and parseResponse
            ...builtQuery,
            // Override the query
            query: gql`
                query Command($id: ID!) {
                    data: Command(id: $id) {
                        id
                        reference
                        customer {
                            id
                            firstName
                            lastName
                        }
                    }
                }`,
        };
    }

    return builtQuery;
}

export default buildGraphcoolProvider({ buildQuery })
```

Reply:
Actually a bad idea as one may need the introspection results for advanced scenarios